compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5674,21 +5703,51 @@ TR::Register *commonStoreEvaluator(TR::Node *node, TR::InstOpCode::Mnemonic op,
      * if valueChild is a compressed refs sequence of address constant NULL,
      * or valueChild is a zero constant integer.
      */
+    bool inZeroReg = false;
+    TR::Register *srcReg;
     if ((valueChildRoot != NULL)
         || (valueChild->getDataType().isIntegral() && valueChild->isConstZeroValue()
             && (valueChild->getRegister() == NULL))) {
-        TR::Register *zeroReg = cg->allocateRegister();
-        generateMemSrc1Instruction(cg, op, node, tempMR, zeroReg);
+        srcReg = cg->allocateRegister();
+        inZeroReg = true;
+    } else {
+        srcReg = cg->evaluate(valueChild);
+    }
+
+    bool canUseSTLR = size <= 8 && tempMR->getUnresolvedSnippet() == NULL;
+    if (sym->isAtLeastOrStrongerThanAcquireRelease() && canUseSTLR) {
```

Comment:
Don't you need `cg->comp()->target().isSMP()` here?
The condition at L.5669-5670 looks like:
```
(cg->comp()->target().isSMP() && sym->isAtLeastOrStrongerThanAcquireRelease()
 && tempMR->getUnresolvedSnippet() != NULL)
```